backend/api/quivr_api/modules/rag_service/rag_service.py

```diff
@@ -84,29 +94,40 @@ def _build_chat_history(
         [chat_history.append(m) for m in transformed_history]
         return chat_history
 
-    async def _build_rag_config(self) -> RAGConfig:
+    async def _get_retrieval_config(self) -> RetrievalConfig:
+        if self.retrieval_config:
+            retrieval_config = self.retrieval_config
+        else:
+            retrieval_config = self._build_retrieval_config()
+
+        return retrieval_config
+
+    async def _build_retrieval_config(self) -> RetrievalConfig:
         model = await self.model_service.get_model(self.model_to_use)  # type: ignore
         api_key = os.getenv(model.env_variable_name, "not-defined")
```

Comment:
Pydantic has `SecretStr` type to encode keys + you can define you model as having parameter that can be loaded from environement variable : 

https://docs.pydantic.dev/latest/concepts/pydantic_settings/#environment-variable-names

Reply:
Not super clear, also I didn't change this part of the function

```python
 model = await self.model_service.get_model(self.model_to_use)  # type: ignore
 api_key = os.getenv(model.env_variable_name, "not-defined")
```